### PR TITLE
[14.0] [ADD] hr_employee_service_product: New module for timesheet repo.

### LINKED ISSUE #423: [RFC][14.0] `hr_employee_service_product` Service Products on Employee Timesheets
# Description

Assign a service product to employees.
This service Product can determine the hour cost, and is stored in the timesheet entries.

This can be useful to model pay grades for employees, both making it easier to set standard costs and to analyze timesheets per grade.

# Configuration

Set the Service Product to be used for each Employee:

1. On the Employees app, open the Employee form
2. On the Employee form, HR Settings tab, set the Service Product field.
3. On the Employee form, HR Settings tab, set the Timesheet Cost to zero if the hour cost should be taken form the Service Product.

# Usage 

When entering timesheet logs, the Service Product will be set on the Product field of the corresponding Analytic Item.
If the Employee Timesheet Cost is set, it will be used, if it is zero, then the unit cost will be taken from the related Service product.

### PR DESCRIPTION
**Pull request by Chandresh-SerpentCS**
Closes #423

### CODE REVIEW

**Comment by Chandresh-SerpentCS** [comment]
@dreispt PR is Ready for your review.

File: hr_employee_service_product/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    "author": "Open Source Integrators, Odoo Community Association (OCA)",
+    "category": "Timesheet",
+    "website": "https://github.com/OCA/timesheet",
+    "depends": ["product", "hr_timesheet"],
+    "data": [
+        "views/hr_employee_views.xml",
+    ],
+    "installable": True,
+    "maintainer": "dreispt",
+    "development_status": "Alpha",
```
Comment (dreispt) [inline hr_employee_service_product/__manifest__.py]:
```suggestion
    "development_status": "Beta",
```

File: hr_employee_service_product/models/hr_employee.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# Copyright (C) 2021 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class HrEmployee(models.Model):
+    _inherit = "hr.employee"
+
+    service_product = fields.Many2one(
```
Comment (dreispt) [inline hr_employee_service_product/models/hr_employee.py]:
```suggestion
    service_product_id = fields.Many2one(
```

File: hr_employee_service_product/models/hr_employee.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+                timesheet.account_id.currency_id,
+                self.env.company,
+                timesheet.date,
+            )
+            result[timesheet.id].update(
+                {
+                    "amount": amount_converted,
+                }
+            )
+        return result
```
Comment (dreispt) [inline hr_employee_service_product/models/hr_employee.py:47]: The Service Product needs to be set in the `product_id` field of the Analytic Item, I don't see that here.

File: hr_employee_service_product/readme/CONTRIBUTORS.rst [hunk 4]
```diff
@@ -0,0 +1,3 @@
+* Daniel Reis <dreis@opensourceintegrators.com>
+* Chandresh Thakkar <cthakkr@opensourceintegrators.com>
+* Shivam Soni <s.soni.serpentcs@gmail.com>
```
Comment (dreispt) [inline hr_employee_service_product/readme/CONTRIBUTORS.rst]:
```suggestion
* `Open Source Integrators <https://opensourceintegrators.com>`_

  * Daniel Reis <dreis@opensourceintegrators.com>
  * Chandresh Thakkar <cthakkr@opensourceintegrators.com>
  * Shivam Soni <s.soni.serpentcs@gmail.com>
```

**Review by dreispt — COMMENTED** [review]
The Service Product needs to be set in the product_id field of the Analytic Item, I don't see that here.

File: hr_employee_service_product/models/hr_employee.py (hunk 3 above)
Comment (Chandresh-SerpentCS) [reply hr_employee_service_product/models/hr_employee.py:47]: Do we need to show the product id field in the timesheet list view?

Comment (dreispt) [reply hr_employee_service_product/models/hr_employee.py:47]: No.

**Comment by dreispt** [comment]
@Chandresh-SerpentCS 
```


Odoo Server Error

Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/addons/base/models/ir_http.py", line 237, in _dispatch
    result = request.dispatch()
  File "/.repo_requirements/odoo/odoo/http.py", line 683, in dispatch
    result = self._call_function(**self.params)
  File "/.repo_requirements/odoo/odoo/http.py", line 359, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/service/model.py", line 94, in wrapper
    return f(dbname, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/http.py", line 347, in checked_call
    result = self.endpoint(*a, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 912, in __call__
    return self.method(*args, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 531, in response_wrap
    response = f(*args, **kw)
  File "/.repo_requirements/odoo/addons/web/controllers/main.py", line 1377, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/.repo_requirements/odoo/addons/web/controllers/main.py", line 1369, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 394, in call_kw
    result = _call_kw_model_create(method, model, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 374, in _call_kw_model_create
    result = method(recs, *args, **kwargs)
  File "<decorator-gen-163>", line 2, in create
  File "/.repo_requirements/odoo/odoo/api.py", line 344, in _model_create_multi
    return create(self, [arg])
  File "/.repo_requirements/odoo/addons/hr_timesheet/models/hr_timesheet.py", line 103, in create
    line._timesheet_postprocess(values)
  File "/.repo_requirements/odoo/addons/hr_timesheet/models/hr_timesheet.py", line 183, in _timesheet_postprocess
    timesheet.write(values_to_write[timesheet.id])
  File "/.repo_requirements/odoo/addons/hr_timesheet/models/hr_timesheet.py", line 114, in write
    result = super(AccountAnalyticLine, self).write(values)
  File "/.repo_requirements/odoo/odoo/models.py", line 3592, in write
    raise ValueError("Invalid field %r on model %r" % (fname, self._name))
Exception

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/http.py", line 639, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/.repo_requirements/odoo/odoo/http.py", line 315, in _handle_exception
    raise exception.with_traceback(None) from new_cause
ValueError: Invalid field 'product_id' on model 'account.analytic.line'

```

**Comment by Chandresh-SerpentCS** [comment]
@dreispt I have fixed the above issue. The Issue was in module dependency.

**Pull request closed without merging**